src/main/kotlin/no/nav/hag/plugins/Security.kt

```diff
@@ -1,34 +1,32 @@
 package no.nav.hag.plugins
 
 
-import io.ktor.server.application.*
-import io.ktor.server.auth.*
+import io.ktor.server.application.Application
+import io.ktor.server.auth.UserIdPrincipal
+import io.ktor.server.auth.authentication
+import io.ktor.server.auth.basic
+import io.ktor.server.auth.bearer
+import no.nav.hag.AuthClient
 import no.nav.hag.Env
-import no.nav.security.token.support.v2.IssuerConfig
-import no.nav.security.token.support.v2.TokenSupportConfig
-import no.nav.security.token.support.v2.tokenValidationSupport
 
-fun Application.configureSecurity(disabled: Boolean = false) {
+fun Application.configureSecurity(authClient: AuthClient, disabled: Boolean = false) {
     if (disabled && Env.isTest()) {
         authentication {
             basic {
                 skipWhen { true }
             }
         }
     } else {
-        val config =
-            TokenSupportConfig(
-                IssuerConfig(
-                    name = "employee",
-                    discoveryUrl = Env.oauth2Environment.wellKnownUrl,
-                    acceptedAudience = listOf(Env.oauth2Environment.clientId),
-                )
-            )
         authentication {
-            tokenValidationSupport(
-                config = config
-            )
+            bearer {
+                authenticate {
+                    if (authClient.introspect(it.token)) {
+                        UserIdPrincipal("hag-admin")
+                    } else {
+                        null
+                    }
+                }
+            }
```

Comment:
Ikke fullstending sikker på at dette er funksjonelt ekvivalent, men i [dokumentasjonen](https://doc.nav.cloud.nais.io/auth/entra-id/how-to/secure/#validate-with-texas) så fremstår det som relativt rett fram.